### PR TITLE
Upgrade imgui, add docking, update imgui example

### PR DESCRIPTION
**Pull request by aquila-const**
Upgrade ImGui to WIP 1.86
Edit supporting libs
Edit imgui_example

### CODE REVIEW

File: core/pen/include/pen.h
Comment (polymonster) [inline core/pen/include/pen.h:49]: how come this needed to change from `u32` to `f32`? Just wondering now about type conversion warnings that may occur and all of the samples initialising this struct needing to change.

Would prefer this not to change in this PR if possible. If there is a reason we can do it in a separate pass.

File: core/put/source/dev_ui.cpp
Comment (polymonster) [inline core/put/source/dev_ui.cpp:333]: if this is no longer needed, delete the whole line, no need keeping in redundant commented out code

File: core/put/source/dev_ui.cpp
Comment (polymonster) [inline core/put/source/dev_ui.cpp:420]: if the enums are now changed can we find out what the new names are and re-instate this commented out code?

File: examples/code/imgui_example/imgui_example.cpp
Comment (polymonster) [inline examples/code/imgui_example/imgui_example.cpp:112]: same as a previous comment, can we remove the commented code or figure out the correct enums to set?

File: core/pen/include/pen.h
Comment (aquila-const) [reply core/pen/include/pen.h:49]: Ahh I thought I had reverted that. It was an update on the ImGui I was confused on will change it back.

**Review by polymonster — CHANGES_REQUESTED** [review]
Mostly looks good, and it will be really nice to have docking inside `pmtech` Just a few comments to address and again I would like to merge into `pmtech/develop` just so I can give a once over test on windows before merging to master.

File: core/put/source/dev_ui.cpp
Comment (aquila-const) [reply core/put/source/dev_ui.cpp:420]: I will do so. I do think they removed it entirely.

File: examples/code/imgui_example/imgui_example.cpp
Comment (aquila-const) [reply examples/code/imgui_example/imgui_example.cpp:112]: Correlating to the changelog, replaced with correct enums.

File: core/pen/include/pen.h
Comment (aquila-const) [reply core/pen/include/pen.h:49]: reverted

File: examples/code/imgui_example/imgui_example.cpp
Comment (polymonster) [inline examples/code/imgui_example/imgui_example.cpp:27]: The .0f isn't necessary anymore since the window_width and height should be u32 again?

**Comment by polymonster** [comment]
Noticing the change to pmbuild submodule, can you confirm if that is up-to-date with pmbuild/master?

**Review by polymonster — COMMENTED** [review]
I'm on the mobile app so it's a bit fiddly. But everything looks in good shape. 

Just the change to .0f and checking pmbuild submodule.

File: examples/code/imgui_example/imgui_example.cpp
Comment (aquila-const) [reply examples/code/imgui_example/imgui_example.cpp:27]: I did change  this let me recheck.

**Comment by aquila-const** [comment]
> Noticing the change to pmbuild submodule, can you confirm if that is up-to-date with pmbuild/master?

This seems to not be up to date. I can pull the latest.

**Review by polymonster — APPROVED** [review]
Nice one, happy for these to merge into develop once the build is successful. 

I will merge develop to master once I get back to a computer but that will probably be Monday now.

Thanks! 😌✌🏻

**Comment by aquila-const** [comment]
> Nice one, happy for these to merge into develop once the build is successful.
> 
> I will merge develop to master once I get back to a computer but that will probably be Monday now.
> 
> Thanks! relieved✌🏻

Will do! I'll take a peek at the other things and also at the trello. I'll try and get something working on the view port as well.

**Pull request merged**
